### DISCUSSION TITLE
Need help clearing localStorage using webdriverio

### DISCUSSION THREAD

**Discussion by salmankhilji** [webdriverio/webdriverio#7089] [Q&A]
The following script fails to clear local storage for a given website:

```js
const { remote } = require('webdriverio');

(async () => {
  const browser = await remote({
    capabilities: {
      browserName: 'chrome',
    },
  });

  await browser.url('https://webdriver.io');

  await browser.execute(() => {
    localStorage.clear();
  });

  await browser.deleteSession();
})();
```

Before running the script, I set the theme to `dark` and expect it to fall back to `light` (the default) after running the script.  However, the theme stays at `dark`.  So that is how I know the storage is not getting cleared.

### Environment

- Ubuntu 20 04 LTS
- Chrome Version 91.0.4472.114 (Official Build) (64-bit)
- node v14.15.5
- package.json: `"webdriverio": "^7.7.4"`

### Additional Info

The issue is that the browser is invoked with a temporary profile path, such as `--user-data-dir=/tmp/lighthouse.v3kZaOa`.

How do I make it so that webdriverio invokes Chrome with my default profile path: `--user-data-dir=~/.config/google-chrome`?

**Comment by salmankhilji** [comment]
For the benefit of those who have similar background as myself, this task is probably better suited to implement directly in puppeteer, like this:

```js
const puppeteer = require('puppeteer-core');

(async () => {
  const browser = await puppeteer.launch({
    executablePath: '/usr/bin/google-chrome-stable',
    userDataDir: '/home/XXXX/.config/google-chrome',
  });
  const page = await browser.newPage();
  await page.goto('https://webdriver.io');
  await page.evaluate(() => {
    window.localStorage.clear();
  });
  await browser.close();
})();
```

No need to use `webdriverio`.

**Comment by christian-bromann** [comment]
> How do I make it so that webdriverio invokes Chrome with my default profile path: `--user-data-dir=~/.config/google-chrome`?

You can pass in Chrome flags via:

```js
capabilities: {
    browserName: 'chrome'
    'goog:chromeArgs': {
        args: ['user-data-dir=~/.config/google-chrome']
    }
}
```

If you use the devtools protocol (no browser driver running), you can also do:

```js
  const browser = await remote({
    capabilities: {
      browserName: 'chrome',
      'wdio:devtoolsOptions': {
          executablePath: '/usr/bin/google-chrome-stable',
          userDataDir: '/home/XXXX/.config/google-chrome',
       }
    },
  });
```

Note that it is recommended to start e2e tests always with a fresh/new chrome profile.
